### PR TITLE
Add mediator integration tests

### PR DESCRIPTION
**Pull request by manoramahp**

### CODE REVIEW

File: modules/integration/src/test/java/org/apache/synapse/mediators/MediatorTestCase.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+     * Loads all configuration info and starts the servers.
+     */
+    public void setUp() {
+        // Print a short intro to the console, so the console output is more readable
+        String title = "Test : " + this.getName();
+        String underline = "";
+        for (int i = 0; i < title.length(); i++) {
+            underline += "=";
+        }
+        System.out.println("\n\n" + title);
```
Comment (nuwanw) [inline modules/integration/src/test/java/org/apache/synapse/mediators/MediatorTestCase.java]: Please remove System.out

File: modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateIDTestCase.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import javax.xml.namespace.QName;
+import java.util.Iterator;
+
+public class IterateIDTestCase extends MediatorTestCase {
+
+    public IterateIDTestCase() {
+        loadConfiguration("/mediators/iterateIDTestConfig.xml");
+    }
+
+    public void testIterateID() throws Exception {
```
Comment (nuwanw) [inline modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateIDTestCase.java]: Method can be removed

File: modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateSOAPActionTestCase.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    public IterateSOAPActionTestCase() {
+        loadConfiguration("/mediators/iterateSOAPActionTestConfig.xml");
+    }
+
+    public void testSOAPAction() throws Exception {
+        String addUrl = "http://localhost:8280/services/iterateWithSOAPActionTestProxy";
+        StockQuoteSampleClient client = getStockQuoteClient();
+        SampleClientResult result = client.requestStandardQuote(addUrl, null, null, "WSO2", null);
+        assertTrue(result.responseReceived());
```
Comment (nuwanw) [inline modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateSOAPActionTestCase.java]: Assert message is missing

File: modules/integration/src/test/java/org/apache/synapse/mediators/switchmediator/SOAPNSBasedSwitchingTestCase.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * Test case for Switch Mediator: Write xpath expression using SOAP 1.1/1.2 NS
+ */
+public class SOAPNSBasedSwitchingTestCase extends MediatorTestCase {
+
+    public SOAPNSBasedSwitchingTestCase() {
+        loadConfiguration("/mediators/soapNSBasedSwitchingTestConfig.xml");
+    }
+
+    public void testXPathOnSOAP11() {
```
Comment (nuwanw) [inline modules/integration/src/test/java/org/apache/synapse/mediators/switchmediator/SOAPNSBasedSwitchingTestCase.java]: method can be redundant

File: modules/integration/src/test/java/org/apache/synapse/mediators/switchmediator/SwitchingBasedOnAddressTestCase.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import org.apache.synapse.mediators.MediatorTestCase;
+import org.apache.synapse.mediators.clients.AxisOperationClient;
+
+public class SwitchingBasedOnAddressTestCase extends MediatorTestCase {
+
+    public SwitchingBasedOnAddressTestCase() {
+        loadConfiguration("/mediators/switchingBasedOnAddressTestConfig.xml");
+    }
+
+    public void testMessageBasedOnAddress() {
```
Comment (nuwanw) [inline modules/integration/src/test/java/org/apache/synapse/mediators/switchmediator/SwitchingBasedOnAddressTestCase.java]: method can be redundant

File: modules/integration/src/test/java/org/apache/synapse/mediators/clients/AxisOperationClient.java [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+    }
+
+    private void init() throws AxisFault {
+        configContext = ConfigurationContextFactory.createConfigurationContextFromFileSystem(
+                clientConfig.getClientRepo(), clientConfig.getAxis2Xml());
+        serviceClient = new ServiceClient(configContext, null);
+    }
+
+    /**
+     * @param trpUrl
```
Comment (sdkottegoda) [inline modules/integration/src/test/java/org/apache/synapse/mediators/clients/AxisOperationClient.java]: Shall we add parameter descriptions?

File: modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateWithMultipleElementsTestCase.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        assertNotNull("Received response is null", response);
+        OMElement soapBody = response.getFirstElement();
+        Iterator iterator =
+                soapBody.getChildrenWithName(new QName("http://services.samples",
+                        "getQuoteResponse"));
+        int count = 0;
+        while (iterator.hasNext()) {
+            count++;
+            OMElement getQuote = (OMElement) iterator.next();
+            assertTrue(getQuote.toString().contains("WSO2 Company"));
```
Comment (sdkottegoda) [inline modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateWithMultipleElementsTestCase.java]: Shall we add an assert message?

File: modules/integration/src/test/java/org/apache/synapse/mediators/MediatorTestCase.java (hunk 1 above)
Comment (manoramahp) [reply modules/integration/src/test/java/org/apache/synapse/mediators/MediatorTestCase.java]: Done

File: modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateIDTestCase.java (hunk 2 above)
Comment (manoramahp) [reply modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateIDTestCase.java]: Done

File: modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateSOAPActionTestCase.java (hunk 3 above)
Comment (manoramahp) [reply modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateSOAPActionTestCase.java]: Added

File: modules/integration/src/test/java/org/apache/synapse/mediators/switchmediator/SOAPNSBasedSwitchingTestCase.java (hunk 4 above)
Comment (manoramahp) [reply modules/integration/src/test/java/org/apache/synapse/mediators/switchmediator/SOAPNSBasedSwitchingTestCase.java]: Removed

File: modules/integration/src/test/java/org/apache/synapse/mediators/switchmediator/SwitchingBasedOnAddressTestCase.java (hunk 5 above)
Comment (manoramahp) [reply modules/integration/src/test/java/org/apache/synapse/mediators/switchmediator/SwitchingBasedOnAddressTestCase.java]: Removed

File: modules/integration/src/test/java/org/apache/synapse/mediators/clients/AxisOperationClient.java (hunk 6 above)
Comment (manoramahp) [reply modules/integration/src/test/java/org/apache/synapse/mediators/clients/AxisOperationClient.java]: Done

File: modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateWithMultipleElementsTestCase.java (hunk 7 above)
Comment (manoramahp) [reply modules/integration/src/test/java/org/apache/synapse/mediators/iterate/IterateWithMultipleElementsTestCase.java]: Done

**Pull request merged**